src/read.c

```diff
@@ -98,32 +95,30 @@ extern inline struct ReaderState * ReaderState(void)
 **  is <Statements>. The functions 'ReadExpr' and 'ReadStats' must therefore
 **  be declared forward.
 */
-static void ReadExpr(ScannerState * s, TypSymbolSet follow, Char mode);
+static void ReadExpr(ReaderState * rs, TypSymbolSet follow, Char mode);
 
-static UInt ReadStats(ScannerState * s, TypSymbolSet follow);
+static UInt ReadStats(ReaderState * rs, TypSymbolSet follow);
 
-static void ReadFuncExprAbbrevSingle(ScannerState * s, TypSymbolSet follow);
+static void ReadFuncExprAbbrevSingle(ReaderState * rs, TypSymbolSet follow);
 
-static void ReadAtom(ScannerState * s, TypSymbolSet follow, Char mode);
+static void ReadAtom(ReaderState * rs, TypSymbolSet follow, Char mode);
 
-static void PushGlobalForLoopVariable(UInt var)
+static void PushGlobalForLoopVariable(ReaderState * rs, UInt var)
 {
-    struct ReaderState * rs = ReaderState();
     if (rs->CurrentGlobalForLoopDepth <
         ARRAY_SIZE(rs->CurrentGlobalForLoopVariables))
         rs->CurrentGlobalForLoopVariables[rs->CurrentGlobalForLoopDepth] = var;
     rs->CurrentGlobalForLoopDepth++;
 }
 
-static void PopGlobalForLoopVariable(void)
+static void PopGlobalForLoopVariable(ReaderState * rs)
 {
-    GAP_ASSERT(ReaderState()->CurrentGlobalForLoopDepth);
-    ReaderState()->CurrentGlobalForLoopDepth--;
+    GAP_ASSERT(rs->CurrentGlobalForLoopDepth);
+    rs->CurrentGlobalForLoopDepth--;
```

Comment:
We also see a lot of changes identical to this: instead of using `ReaderState()`, we now use a `ReaderState` instance named `rs` which the callers passes down to us.